utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -383,20 +379,56 @@ pub enum Command {
 	/// initializes the state from the remote node, and starts applying that block, plus all the
 	/// blocks that follow, to the same growing state.
 	FollowChain(commands::follow_chain::FollowChainCmd),
+
+	/// Create a new snapshot file.
+	CreateSnapshot(commands::create_snapshot::CreateSnapshotCmd),
+}
+
+#[derive(Debug, Clone)]
+enum Runtime {
+	/// Use the given path to the wasm binary file.
+	///
+	/// It must have been compiled with `try-runtime`.
+	Path(PathBuf),
+
+	/// Use the code of the remote node, or the snapshot.
+	///
+	/// In almost all cases, this is not what you want, because the code in the remote node does
+	/// not have any of the try-runtime custom runtime APIs.
+	Existing,
+}
+
+impl FromStr for Runtime {
+	type Err = String;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		Ok(match s.to_lowercase().as_ref() {
+			"existing" => Runtime::Existing,
+			x @ _ => Runtime::Path(x.into()),
+		})
+	}
 }
 
 /// Shared parameters of the `try-runtime` commands
 #[derive(Debug, Clone, clap::Parser)]
 #[group(skip)]
 pub struct SharedParams {
 	/// Shared parameters of substrate cli.
+	///
+	/// TODO: this is only needed because try-runtime is embedded in the substrate CLI. It should
+	/// go away.
 	#[allow(missing_docs)]
 	#[clap(flatten)]
-	pub shared_params: sc_cli::SharedParams,
+	shared_params: sc_cli::SharedParams,
 
-	/// The execution strategy that should be used.
-	#[arg(long, value_name = "STRATEGY", value_enum, ignore_case = true, default_value_t = ExecutionStrategy::Wasm)]
-	pub execution: ExecutionStrategy,
+	#[arg(long)]
+	/// The runtime to use.
+	///
+	/// Must be a path to a wasm blob, compiled with `try-runtime` feature flag.
+	///
+	/// Or, `existing`, indicating that you don't want to overwrite the runtime. This will use
+	/// whatever comes from the remote node, or the snapshot file.
```

Comment:
```suggestion
	/// The runtime to use.
	///
	/// Must be a path to a wasm blob, compiled with `try-runtime` feature flag.
	///
	/// Or, `existing`, indicating that you don't want to overwrite the runtime. This will use
	/// whatever comes from the remote node, or the snapshot file.
	#[arg(long)]
```